### PR TITLE
Update SPEIbase Dataset to Version 2.9

### PR DESCRIPTION
**Pull request by B0RJA**
This pull request aims to update the SPEIbase dataset to version 2.9, incorporating an extended coverage up to the year 2023.

### CODE REVIEW

File: catalog/CSIC/CSIC_SPEI_2_9.jsonnet [hunk 1]
```diff
@@ -1,5 +1,5 @@
-local id = 'CSIC/SPEI/2_8';
```
Comment (simonff) [inline catalog/CSIC/CSIC_SPEI_2_9.jsonnet:1]: could you open a bug in https://b.corp.google.com/issues?q=status:(open%20%7C%20new%20%7C%20assigned%20%7C%20accepted)%20componentid:1161653&p=1 for tracking this?

Note that we are now more open to adding to the catalog datasets directly from source asset paths (under projects/speibase in your case). Would you like to do that?

**Comment by B0RJA** [comment]
I have create the bug:
https://issuetracker.google.com/issues/298556739

What do you mean by adding the dataset directly from my source path? I don't understand the difference with the method we followed for the initial release of the data (bug in issuetracker + pull-request in this repository).

This is the source path of the updated dataset:
projects/speibase/assets/SPEIv29

File: catalog/CSIC/CSIC_SPEI_2_9.jsonnet [hunk 2]
```diff
@@ -431,8 +431,8 @@ local self_ee_catalog_url = ee_const.ee_catalog_url + basename;
   'sci:citation': |||
```
Comment (simonff) [inline catalog/CSIC/CSIC_SPEI_2_9.jsonnet:431]: could you please add
'gee:skip_indexing': true
(example: https://github.com/google/earthengine-catalog/blob/main/catalog/GOOGLE/GOOGLE_GLOBAL_CCDC_V1.jsonnet#L15)

We need this to be set until the catalog images are generated.

File: catalog/CSIC/catalog.jsonnet [hunk 3]
```diff
@@ -18,6 +18,6 @@ local self_url = base_url + base_filename;
     ee.link.root(),
     ee.link.parent(parent_url),
     ee.link.self_link(self_url),
-    ee.link.child_collection('CSIC_SPEI_2_8', base_url),
+    ee.link.child_collection('CSIC_SPEI_2_9', base_url),
```
Comment (simonff) [inline catalog/CSIC/catalog.jsonnet:21]: don't change this file yet while we have gee:skip_indexing set.

**Review by simonff — APPROVED** [review]

**Pull request merged**